app/gameboard_api.py

```diff
@@ -144,3 +205,28 @@ def _adjust_blue_points(self, red_link):
         if red_link['visibility']['score'] >= 50:
             return dict(value=-2, reason='high visibility {} team activity not detected'.format(self.RED_TEAM))
         return dict(value=-1, reason='low visibility {} team activity not detected'.format(self.RED_TEAM))
+
+    @staticmethod
+    async def _construct_splash_tactics(abilities):
+        tactics = dict()
+        for ab in abilities:
+            technique = (ab.technique_id, ab.technique_name)
+            if ab.tactic in tactics:
+                tactics[ab.tactic.lower()].add(technique)
+            else:
+                tactics[ab.tactic] = set([technique])
+        for tactic in tactics:
+            tactics[tactic] = list(tactics[tactic])
+        return tactics
+
+    async def _construct_hidden_red_operations(self):
+        hidden_ops = await self.data_svc.locate('operations', match=dict(access=self.Access.HIDDEN))
+        hidden_red_ops = []
+        for op in hidden_ops:
+            if op.group and op.group != 'blue':
```

Comment:
this should probably be based on Access instead of group name.  Someone could start a blue agent (in the blue group) and then change it's group to be something else (e.g. "blue-windows", "blue-endpoints").  I think it's somewhat likely that we might try to endorse using >1 group of blue agents someday.

Reply:
In the training use case, operations are going to be provided hidden. So maybe instead we can look at the access of the actual agent.